blinding.go

```diff
@@ -0,0 +1,129 @@
+package sphinx
+
+import (
+	"fmt"
+
+	"github.com/btcsuite/btcd/btcec/v2"
+)
+
+const routeBlindingHMACKey = "blinded_node_id"
+
+// BlindedPath represents all the data that the creator of a blinded path must
+// transmit to the builder of route that will send to this path.
+type BlindedPath struct {
+	// IntroductionPoint is the real node ID of the first hop in the blinded
+	// path. The sender should be able to find this node in the network
+	// graph and route to it.
+	IntroductionPoint *btcec.PublicKey
+
+	// BlindingPoint is the first ephemeral blinding point. This is the
+	// point that the introduction node will use in order to create a shared
+	// secret with the builder of the blinded route. This point will need
+	// to be communicated to the introduction node by the sender in some
+	// way.
+	BlindingPoint *btcec.PublicKey
+
+	// BlindedHops is a list of ordered blinded node IDs representing the
+	// blinded route. Note that the first node ID is the blinded node ID of
+	// the introduction point which does not strictly need to be transmitted
+	// to the sender.
+	BlindedHops []*btcec.PublicKey
+
+	// EncryptedData is a list of encrypted_data byte arrays. Each entry
+	// has been encrypted by the blinded route creator for a hop in the
+	// blinded route.
+	EncryptedData [][]byte
```

Comment:
A nice to have here would be a `BlindedPair` struct combining nodeid + data, possibly aliasing `BlindedPathHop` ?

Using this code for onion messages, I've found that there are a few cases where you want to sling the combination around and using `BlindedPathHop` directly seems ambiguous because it's currently only used for  the un-blinded data.

Reply:
yeah, really good point. Let me know what you think of the latest updates 👍